### ISSUE TITLE
Log messages in OpenGaze / GazePoint

### ISSUE THREAD

**Issue by smathot** [esdalmaijer/PyGaze#165]
As I mentioned in #164 , the current log-file format for OpenGaze / GazePoint does not lend itself well to logging lots of information. Essentially, log messages are recorded as part of the `USER_DATA` field of gaze samples. Therefore, you can only have one log message per sample (i.e. 60 log messages per second). And what's worse: users lose data when they send log messages more often than that.

This limitation isn't necessary, because the log rate of PyGaze isn't limited by the sampling rate of the eye tracker. The main question is what we would like the log file to look like if there are multiple messages. Do you have any suggestions?

**Comment by esdalmaijer** [comment]
Yes, I elected to use the `USER` column because this seemed to be how GazePoint's OpenGaze API preferred logging.

To tackle the problem, I've tried an internal queue of messages that are simply stacked together until we know for certain that they are logged. A problem with this approach is that it requires monitoring the incoming logs for messages, and there seems to be a delay on this sometimes. Haven't fully figured out why, to be honest, but the net result was that sometimes messages repeated a few times (and some still slipped through the cracks!).

For other trackers, I've used markers at the start of lines for data (e.g. DAT) or a log messages (e.g. MSG). We could do something similar here, and simply bind them to the latest incoming sample. (I.e. use the latest incoming sample time as a timestamp for the logged message.)

**Comment by smathot** [comment]
> We could do something similar here, and simply bind them to the latest incoming sample. (I.e. use the latest incoming sample time as a timestamp for the logged message.)

Yes, that probably seems like the best solution. So basically messages correspond to additional rows (one per message) that are timestamped with the latest sample. What would that look like in the log file?

**Comment by esdalmaijer** [comment]
Previously, I used to do something like:

`f.write("\n" + ",".join(["MSG", t, msg]))`

But after seeing some common text-loading functions choke on this (different number of columns per line), I now think it should be something like this:

`f.write("\n" + ",".join(["MSG", t, msg] + (n_cols-3)*[""]))`

**Comment by JeanBaptisteMelmi** [comment]
Hi Sebastiaan and Edwin. I'm a bit stuck here with the gazepoint. I'm sending log messages by using the "eyetracker.log" command into an inline script. Even if I'm sending messages with several seconds of interval, in the .tsv file, at least half of the messages are missing... Do you have any ideas ?
